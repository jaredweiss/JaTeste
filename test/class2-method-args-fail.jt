func int main()
{
	struct circle *my_circle;
	int diameter;
	int i;
	my_circle = new struct circle;
	my_circle->set_radius(10);

	diameter = my_circle->get_diameter();
	print(diameter);

	my_circle->set_radius(10,1);

	return 0;
}

struct circle {
	int radius;
	int diameter;
	
	method void set_radius(int c)
	{
		radius = c;
		diameter = radius * 2;
	}

	method int get_radius()
	{
		return radius;
	}	

	method int get_diameter()
	{
		return diameter;
	}


};
