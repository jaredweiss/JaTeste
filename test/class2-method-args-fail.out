Scanned
Parsed
Fatal error: exception Exceptions.InvalidArgumentsToFunction("circleset_radius is supplied with wrong args")
